ocfweb/docs/docs/services/hpc.md

```diff
@@ -0,0 +1,160 @@
+the Student Tech Fund
+
+## Slurm
+
+We currently use [Slurm][slurm] as our workload manager for the cluster. We
+will soon post technical documentation about our Slurm configuration, but briefly,
+Slurm is a free and open-source job scheduler which helps distribute jobs from
+all users evenly among HPC computers, referred to as nodes. In short, all of
+your programs will be run through Slurm. To use Slurm there are several commands
+that will be helpful:
+
+* `srun`: Used to submit jobs.
+* `scontrol`: Used to view and edit the Slurm configuration.
+* `squeue`: Used to view running and queued jobs.
+
+## Dependencies
+For managing application dependencies, you currently have two options:
+
+### Virtual Environments
+
+First you can use a [virtual environment][venv] if you are using Python
+packages. To create a virtual environment navigate to your home directory
+and run the following commands:
+
+```
+virtualenv -p python3 venv
+. venv/bin/activate
+```
+
+This will allow you to *pip install* any Python packages that the OCF does not
+already have for your program.
+
+### Singularity
+
+For those who need access to non-Python dependencies or have already integrated
+their program into Docker, the second option is to use Singularity containers.
+[Singularity][singularity] is a containerization platform developed at Lawrence
+Berkeley National Laboratory that is designed specifically for HPC environments.
+To read more about the benefits of Singularity you can look
+[here][singularity_article]. We suggest a particular workflow, which will help
+simplify deploying your program on our infrastructure.
+
+#### Installing
+
+We recommend that you do your development on our HPC infrastructure, but you
+can also develop on your own machine if you would like. If you are running
+Linux on your system, you can install Singularity from the official apt repos:
+
+```
+sudo apt install singularity-container
+```
+
+If you are running Mac you can look [here][mac_install], or Windows [here][win_install].
+
+#### Building Your Container
+
+```
+singularity build --sandbox ./my_container docker://ubuntu
+```
+This will create a Singularity container named `my_container`. If you are still
+testing out your container on your system, we suggest you use the `--sandbox`
+option. This will allow you to install new packages in your container as you
+need them. The `docker://ubuntu` option notifies Singularity to bootstrap the
+container from the official Ubuntu docker container on [Docker Hub][docker_hub].
+There is also a [Singularity Hub][singularity_hub], from which you can directly
+pull Singularity images in a similar fashon. We also have some pre-built
+containers that you may use to avoid having to build your own. They are
+currently located at `/home/containers` on the Slurm master node.
+
+#### Using Your Container
+
+```
+singularity shell my_container
+```
+The above command will allow you to shell into your container. By default your
```

Comment:
Maybe explain what that does? Like what exactly would change between a shell in the host system and a shell in the container.

Reply:
I think for the sake of keeping this documentation clear it may be simpler to describe it as if it were a different system, since it would likely be confusing to explain namespaces and other containerization tech. Perhaps this would be good to include in the singularity technical docs when we make them?